src/jestUtils.ts

```diff
@@ -0,0 +1,270 @@
+import { fireEvent } from '@testing-library/react-native';
+
+export const fireGestureHandlerEvent = (
+  component: any,
+  name: string,
+  config: any
+) => {
+  fireEvent(component, name, config);
+};
+
+export interface TapConfig {
+  x?: number;
+  y?: number;
+  absoluteX?: number;
+  absoluteY?: number;
+}
+
+export interface PanConfig extends TapConfig {
+  translationX?: number;
+  translationY?: number;
+  velocityX?: number;
+  velocityY?: number;
+}
+
+export interface LongPressConfig extends TapConfig {
+  duration?: number;
+}
+
+export interface RotationConfig {
+  rotation?: number;
+  velocity?: number;
+  anchorX?: number;
+  anchorY?: number;
+}
+
+export interface PinchConfig {
+  scale?: number;
+  velocity?: number;
+  focalX?: number;
+  focalY?: number;
+}
+
+const sendBeginEvent = (component: any, eventData?: Record<string, any>) => {
+  // simulate undetermined -> begin state change
+  fireEvent(component, 'gestureHandlerStateChange', {
+    nativeEvent: {
+      oldState: 0,
+      state: 2,
+      ...eventData,
+    },
+  });
+};
+
+const sendProgressEvent = (component: any, eventData?: Record<string, any>) => {
+  // simulate begin -> active state change
+  fireEvent(component, 'gestureHandlerStateChange', {
+    nativeEvent: {
+      oldState: 2,
+      state: 4,
+      ...eventData,
+    },
+  });
+};
+
+const sendEndEvent = (component: any, eventData?: Record<string, any>) => {
+  // simulate active -> end state change
+  fireEvent(component, 'gestureHandlerStateChange', {
+    nativeEvent: {
+      oldState: 4,
+      state: 5,
+      ...eventData,
+    },
+  });
+};
+
+const runEventsSequence = (
+  component: any,
+  eventData?: Record<string, any>,
+  configBegin?: Record<string, any>,
+  configProgress?: Record<string, any> | Record<string, any>[],
+  configEnd?: Record<string, any>
+) => {
+  sendBeginEvent(component, { ...eventData, ...configBegin });
+  if (Array.isArray(configProgress)) {
+    for (const item of configProgress) {
+      sendProgressEvent(component, { ...eventData, ...item });
+    }
+  } else {
+    sendProgressEvent(component, { ...eventData, ...configProgress });
+  }
```

Comment:
Only the first item from `configProgress` should be sent as `gestureHandlerStateChange` event.
```suggestion
  if (Array.isArray(configProgress)) {
    for (const item of configProgress) {
      if (item === configProgress[0]) {
        sendActivateEvent(component, { ...eventData, ...item });
      } else {
        sendProgressEvent(component, { ...eventData, ...item });
      }
    }
  } else {
    sendActivateEvent(component, { ...eventData, ...configProgress });
  }
```